src/runtime/write_debug_image.cpp

```diff
@@ -125,6 +128,42 @@ struct ScopedFile {
     }
 };
 
+#if (defined(__BYTE_ORDER) && __BYTE_ORDER == __BIG_ENDIAN) || defined(HALIDE_FORCE_BIG_ENDIAN)
```

Comment:
There are parts of the compiler that assume the target is little-endian already, so I don't think you need to worry about big-endian systems in the runtime.